app/views/plannings/base/update.js.haml

```diff
@@ -8,12 +8,9 @@ App.plannings.selectable.clear();
 App.plannings.selectable.destroy();
 $('#weeks_header_#{dom_id(@board.subject)}').html('#{j(render 'calendar_weeks')}');
 $('#planned_#{dom_id(@board.subject)}').html('#{j(render('header_planned'))}');
-- @board.rows.each do |(employee_id, work_item_id), items|
-  - row = j(render('row',
-                   items: items,
-                   legend: @board.row_legend(employee_id, work_item_id),
-                   row_total: @board.total_row_planned_hours(employee_id, work_item_id)))
-  $('##{planning_row_id(employee_id, work_item_id)}').html('#{row}');
+- @board_updates.each do |row_data|
+  - row = j(render('row', **row_data[:locals]))
+  $('##{planning_row_id(*row_data[:ids])}').html('#{row}');
```

Comment:
I've moved the logic from here to the controller, so it doesn't grow more and more in the view.